presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -490,6 +490,29 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                     combineConjuncts(postJoinConjuncts.build()));
         }
 
+        private boolean shouldRemoveOuterJoinConjunct(Collection<Symbol> outerSymbols, Expression conjunct)
+        {
+
+            if (!(conjunct instanceof ComparisonExpression)) {
```

Comment:
ops, it's ok probably, since you are using `innerInference` which doesn't use `joinPredicate`, so by removing it, you don't change semantic.